service/src/main/java/bio/terra/profile/service/profile/flight/create/CreateProfileVerifyAccountStep.java

```diff
@@ -17,15 +17,16 @@
 record CreateProfileVerifyAccountStep(
     CrlService crlService, BillingProfile profile, AuthenticatedUserRequest user) implements Step {
 
+  public static List<String> PERMISSIONS_TO_TEST =List.of("billing.resourceAssociations.create");
+
   @Override
   public StepResult doStep(FlightContext context) throws InterruptedException {
     var billingCow = crlService.getBillingClientCow(user);
 
-    var permissionsToTest = List.of("billing.resourceAssociations.create");
     var testPermissionsRequest =
         TestIamPermissionsRequest.newBuilder()
-            .setResource(BillingAccountName.of(profile.billingAccountId().get()).toString())
-            .addAllPermissions(permissionsToTest)
+            .setResource(BillingAccountName.of(profile.getRequiredBillingAccountId()).toString())
```

Comment:
This change trues up the exception thrown here to be MissingRequiredFieldsException; previously it was throwing IllegalArgumentException due to de-referencing a null optional, which was different than the exception we are throwing on the similar Azure codepath.